testConnector/src/test/java/io/vantiq/extsrc/testConnector/TestTestConnectorCore.java

```diff
@@ -127,6 +126,30 @@ public void testProcessRequest() {
         assert resultEnvVars.get(NONEXISTENT_ENV_VAR) == null;
     }
 
+    @Test
+    public void testRawBytes() {
+        assumeTrue(filename != null && fileContents != null);
+
+        // Put data in the request and call the process method
+        Map<String, Object> request = new LinkedHashMap<>();
+        List<String> filenames = new ArrayList<>();
+        filenames.add(filename);
+        request.put("filenames", filenames);
+        request.put("rawBytes", true);
+        Map result = core.processRequest(request, null);
+
+        // Check that the data is base64 encoded version of what we expect
+        assert result.get("files") instanceof Map;
+        Map resultFiles = (Map) result.get("files");
+        assert resultFiles.get(filename) instanceof String;
+        String fileString = (String) resultFiles.get(filename);
+        assert !fileString.isEmpty();
+        assert !fileString.equals(fileContents);
+        String decodedString = new String(Base64.getDecoder().decode(fileString.getBytes()));
```

Comment:
Why is this String as opposed to bytes?

Reply:
The `fileString` is the base64 encoded version of the bytes, and then I test that things actually work by showing that the base64 version is not equal to the contents of the file, but once decoded they match up.

Reply:
I see what it's doing but you're passing strings as binary, I guess.  Why not just deploy some "real" binary data & check against that. So the fileString coming back is a base64 encoded string -- got that.  But I don't understand why `decodedString` is a `String`?  Why not just get the bytes & compare.

And, even with all that, why must we string the newlines out of the decoded stuff.  Decode(encode(orig)) == orig should always be true.